README.md

```diff
@@ -27,78 +26,78 @@ Interface Exporter Action automates the process of extracting TypeScript interfa
 
 ## Example
 
-Interface Exporter Action generates an NPM package with your interfaces ABIs:
+Interface Exporter Action generates NPM packages with your interfaces and contracts ABIs using a matrix of arguments with both and then publishes them to NPM:
 
 ```yaml
-- name: Use Node.js
-  uses: actions/setup-node@v3
-  with:
-    node-version: 16
-    registry-url: "https://registry.npmjs.org"
-
-- name: Install dependencies
-  run: yarn --frozen-lockfile
-
-- name: Build project and generate out directory
-  run: yarn build
-
-- name: Export interfaces for abi compatibility
-  uses: defi-wonderland/interface-exporter-action@v1
-  with:
-    out_dir: ./out
-    interfaces_dir: src/interfaces
-    typing_type: abi
-    package_name: @my-cool-protocol/core-interfaces-abi
-    destination_dir: abi-project
-
-- name: Publish
-  run: cd abi-project && npm publish --access public
-  env:
-    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-```
-
-Interface Exporter Action generates an NPM package with your contracts:
-
-```yaml
-- name: Use Node.js
-  uses: actions/setup-node@v3
-  with:
-    node-version: 16
-    registry-url: "https://registry.npmjs.org"
-
-- name: Install dependencies
-  run: yarn --frozen-lockfile
-
-- name: Build project and generate out directory
-  run: yarn build
-
-- name: Export contracts, abis and interfaces
-  uses: defi-wonderland/interface-exporter-action@v1
-  with:
-    out_dir: ./out
-    interfaces_dir: src/interfaces
-    contracts_dir:  src/contracts
-    typing_type: contracts
-    package_name: @my-cool-protocol/core-interfaces-abi
-    destination_dir: abi-project
-
-- name: Publish
-  run: cd abi-project && npm publish --access public
-  env:
-    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+name: Export And Publish Interfaces And Contracts
+
+on: [push]
+
+jobs:
+  export:
+    name: Generate Interfaces And Contracts
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        export_type: ['interfaces', 'contracts']
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        run: yarn --frozen-lockfile
+
+      - name: Build project and generate out directory
+        run: yarn build
+
+      - name: Update version
+        run: yarn version --new-version "0.0.0-${GITHUB_SHA::8}" --no-git-tag-version
+
+      - name: Export Solidity - ${{ matrix.export_type }}
+        uses: defi-wonderland/interface-exporter-action@latest
```

Comment:
Is it ok to use `@latest`?

Reply:
@0xDiscotech I believe it will work only if we tag a release as latest. Not sure tbh, better specify @v1 or something like that. Also as a user, I'd prefer setting a known version to avoid the action stealing the NPM token or rugging me in a way, which is possible if you're always pulling the latest version.

Reply:
Good to know! Will use @v1 so